### PR TITLE
feat: [PL-29440]: Generic forced delete support and its adoption in connector

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
### Summary

- Made a common hook to integrate force delete for any entity
- Integrated it for connector
- Added feature flag PL_FORCE_DELETE_CONNECTOR_SECRET 
- Hook can also be used to show delete error even if force delete is not supported.
- Fixed one of the connectors list skipped test

[Mocks can be found here.](https://www.figma.com/file/DFFQhhMCRvJKXctDVCUfP9/Connectors?node-id=1017%3A148312)

#### Screenshots
**Before changes**

Screenshot 2022-11-15 at 1 59 30 PM

**After change**
Screenshot 2022-11-15 at 1 58 46 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/useEntityDeleteErrorHandlerDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  return (
+    <Layout.Vertical>
+      <Text margin={{ bottom: 'medium' }} font={{ variation: FontVariation.BODY2_SEMI }}>
+        {getString('common.referenceTextWarning', {
+          type: typeLabelText,
+          name
+        })}
+      </Text>
+      {props.forceDeleteCallback ? (
+        <Formik onSubmit={noop} formName={'forcedDelete'} initialValues={{ forcedDelete: false }}>
```
[the remark this answers is not in the document]
Comment (sunnykesh-harness) [reply src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/useEntityDeleteErrorHandlerDialog.tsx:69]: Will update while integrating it with secret.

File: src/modules/10-common/strings/strings.en.yaml [hunk 2]
```diff
@@ -433,6 +433,10 @@ errorHandler:
   communityForum: ' Harness Community Forum'
 noRefData: 'No references found'
 references: 'References'
+referenceButtonText: Go view references
```
[the remark this answers is not in the document]
Comment (sunnykesh-harness) [reply src/modules/10-common/strings/strings.en.yaml:436]: It is as in mock. Let me put this to PM.

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    setForcedDeleteEnabledMock.mockReset()
+    redirectToReferencedByMock.mockReset()
+    closeDialogMock.mockReset()
+  })
+
+  test('should render error without forcedDeleteSupport', async () => {
+    const { queryByText } = render(
+      <TestWrapper
+        path={routes.toConnectors({
+          accountId: ':accountId'
```
Comment (reetika04) [inline src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx:39]: You can use accountPathProps from @common/utils/routeUtils

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      >
+        <CustomButtonContainer
+          closeDialog={noop}
+          redirectToReferencedBy={redirectToReferencedByMock}
+          forceDeleteCallback={forceDeleteCallbackMock}
+          forcedDeleteEnabled={true}
+        />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
```
Comment (reetika04) [inline src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx:122]: Can we avoid snapshot testing?

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        <ContentText
+          entity={commonProp.entity}
+          forceDeleteCallback={forceDeleteCallbackMock}
+          forcedDeleteEnabled={false}
+          setForcedDeleteEnabled={setForcedDeleteEnabledMock}
+        />
+      </TestWrapper>
+    )
+
+    expect(container).toMatchSnapshot()
```
Comment (reetika04) [inline src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx:71]: Avoid snapshot testing

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/useEntityDeleteErrorHandlerDialog.tsx (hunk 1 above)
Comment (reetika04) [inline src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/useEntityDeleteErrorHandlerDialog.tsx:69]: If there is only one entry and no validation can we avoid formik? You can directly use Checkbox.

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx (hunk 4 above)
Comment (sunnykesh-harness) [reply src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/__tests__/useEntityDeleteErrorHandlerDialog.test.tsx:122]: IMO we should not completely avoid snapshot testing. We should avoid long and duplicated snapshots and have enough assertion testing along with it.

File: src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/useEntityDeleteErrorHandlerDialog.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/10-common/hooks/EntityDeleteErrorHandlerDialog/useEntityDeleteErrorHandlerDialog.tsx:69]: Good point, will do clean up in follow up PR.

**Review by mayankVermaHarness — APPROVED** [review]
Please take care of pending comments from @abhinavrastogi-harness / @reetika04 in a new PR

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**